elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchBackendQueueProcessor.java

```diff
@@ -24,10 +35,12 @@
 public class ElasticsearchBackendQueueProcessor implements BackendQueueProcessor {
 
 	private ElasticsearchIndexManager indexManager;
+	private ExtendedSearchIntegrator searchIntegrator;
 
 	@Override
 	public void initialize(Properties props, WorkerBuildContext context, IndexManager indexManager) {
 		this.indexManager = (ElasticsearchIndexManager) indexManager;
+		this.searchIntegrator = context.getUninitializedSearchIntegrator();
```

Comment:
> The SI in the index manager is still null at this point, so this doesn't work.

I sure know it now that I made the mistake the other day :). But when you use it in the apply* methods, it is, so you could have used the indexManager.searchIntegrator without initializing a specific one in the constructor.

Either this or your subsequent PR is fine IMHO.